thunder/core/transforms.py

```diff
@@ -3746,7 +3757,6 @@ def backward_fn(saved_for_backward, cotangents):
             out = tree_flatten(out)[0]
         return out
 
-    saved_for_backward = forward_trace.output[1]
     backward_trace = construct_trace(rename_proxies=False)(backward_fn, saved_for_backward, cotangents)
```

Comment:
We're lucky that rename_proxies=False is used here. Otherwise, all of the "saved_for_backward" proxies would be new proxies with new names and the approach wouldn't work.

Reply:
Just for my education. Would you mind elaborate on how new names would affect this PR? I though even with renaming, the sequence/order of saved_for_backward would still remain the same and what I did here is just replacing according to their index. i.e. whether a proxy has been renamed or not, wouldn't really affect the code change.

Reply:
You're right, I read the code incorrectly.